seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/TableSchema.java

```diff
@@ -20,25 +20,37 @@
 import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
 import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
 
-import lombok.AllArgsConstructor;
+import lombok.AccessLevel;
 import lombok.Data;
+import lombok.Getter;
 
 import java.io.Serializable;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /** Represent a physical table schema. */
 @Data
-@AllArgsConstructor
 public final class TableSchema implements Serializable {
     private static final long serialVersionUID = 1L;
     private final List<Column> columns;
```

Comment:
Sadly, the columns will be changed in other place, for example: https://github.com/apache/seatunnel/blob/97a1b00e481d23314a9b498556653ab34d8764c7/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/sink/AbstractJdbcSinkWriter.java#L92
It will let columns and columnNames size not match.

Reply:
fixed